howso/feature_residuals.amlg

```diff
@@ -1063,6 +1142,142 @@
+						compressed_values
+							(values (append compressed_values (indices !sharedDeviationsMap)) (true))
+					))
+					;if the compressed_values is not an list, it is an assoc.
+					(let
+						(assoc
+							;features that were dropped and had their values combined into their primary shared deviations feature
+							removed_features
+								(filter
+									(lambda
+										(not (contains_value
+											(indices compressed_values)
+											(current_value)
+										))
+									)
+									(indices !sharedDeviationsMap)
+								)
+						)
+						;grab the values for the removed features by looking up their primary key values
+						(declare (assoc
+							removed_features_completed
+								(map
+									(lambda
+										(get compressed_values (get !sharedDeviationsMap (current_index)))
+									)
+									(zip removed_features)
+								)
+						))
+						(assign (assoc
+							compressed_values
+								(append compressed_values removed_features_completed)
+						))
+					)
+				)
+			)
+			compressed_values
+		)
+
+	;Helper function to prep shared feature deviations, by taking in a list of features and their list of residuals,
+	; combining all of the residuals under the primary feature key of the group of the shared deviations, and removing the features
+	; from the features list whose residuals have been combined into another features.
+	;parameters:
+	; features_list: An assoc where the key is the variable name under which the prepped features list will be stored after pre, the value is the feature list to be prepped.
+	; feature_residuals_lists: An assoc where the key is the variable name under which the prepped features residuals list will be stored after prep, the value is the feature residuals lists to be prepped.
+	#!PrepSharedDeviations
+	(declare
+		(assoc
+			features_list (assoc)
+			feature_residuals_lists (assoc)
+		)
+
+		(declare (assoc
+			joined_original_features_residuals_list
+				(zip (first (values features_list)) (first (values feature_residuals_lists)))
+		))
+
+		;assoc of features and their residuals that are not in a shared deviations group
+		(declare
+			(assoc non_shared_deviations_feaures_list
+				(remove joined_original_features_residuals_list (indices !sharedDeviationsMap))
+			)
+		)
+
+		;assoc of primary shared deviations features and their residuals that are combined from all the other features in their shared deviations group
+		(declare (assoc
+			shared_deviations_residuals_list_combined
+				(map
+					(lambda
+						(let
+							(assoc
+								;the group of features in the shared deviations group
+								feature_group (call !GetSharedDeviationGrouping (assoc feature_group_to_retrieve (current_index 2)))
+								current_feature (current_index 1)
+							)
+							(reduce
+								(lambda (append (previous_result) (current_value)) )
+								(map
+									(lambda (get joined_original_features_residuals_list (current_value)))
+									feature_group
+								)
+							)
```

Comment:
we joked, but I do think this might be the exact same as a (apply "append" ...). I'd much prefer that imo (looks simpler to me at least) so please try it and change it to that if it works lol